templatestore/frontend/src/components/Tinyurlcomp.js

```diff
@@ -0,0 +1,199 @@
+import React, { useState, useEffect } from 'react';
+import styles from '../style/templateScreen.less';
+import axios from 'axios';
+import { backendSettings } from './../utils.js';
+import Dropdowncomp from './Dropdowncomp';
+export default function Tinyurlcomp(props) {
+    if (props.contextData === '') return <></>;
+    console.log(props.listOfUrls);
+    var tinyUrlObjLocal = props.tinyUrlObj;
+    const [items, setItems] = useState([{ urlKey: "", expiry: "" }]);
+    const [visited, setVisited] = useState({});
+    useEffect(() => {console.log(props.tinyUrlObj); populateItems(props.tinyUrlObj);}, [props.tinyUrlObj]);
+    function populateItems(tinyUrlObj) {
+        if (tinyUrlObj.length === 0) return;
+        let temp = [...items];
+        let tempvisited = { ...visited };
+        tinyUrlObj.forEach(obj => {
+            tempvisited[obj['urlKey']] = 1;
+            temp.splice(temp.length - 1, 0, {
+                urlKey: obj['urlKey'],
+                expiry: obj['expiry'] });
+        });
+        setVisited(tempvisited);
+        setItems(temp);
+    }
+    function addNewDropdown() {
+        if (items.length) {
+            if (items[items.length - 1]['urlKey'] === "" || items[items.length - 1]['expiry'] === "") {
+                props.showAlerts("Please select valid values first");
+                return;
+            }
+        }
+        var newItems = [...items, { urlKey: "", expiry: "" }];
+        setItems(newItems);
+    }
+    function removeDropdown(e, givenId) {
+        e.preventDefault();
```

Comment:
Is it required to prevent any default behavior here?

Reply:
I guess yes, because after rerendering, the value of e.target.value was coming out to be null.